resources.md

```diff
@@ -32,8 +32,26 @@ It's often said that the best way to learn to write programs is to write program
 
-## Resources that cost money
+## Sites that cost money
 
 - [Python Morsels](https://www.pythonmorsels.com/): A short Python exercise every week, followed a fews days later by a number of idiomatic solutions, with explanations of each one.
 
 - [PyBites](https://codechalleng.es/): <q>Hone your Python skills by coding and verifying exercises in the comfort of your browser.</q> Offers a small number of free exercises, but most require buying a bundle or monthly subscription. Exercises are grouped by skill level (Bites of Py) and topic (Learning Paths).
+
+## Books
+
+There are thousands of books on programming in Python. This list focuses on titles helpful for beginners and advanced beginners. Most of the titles are available for free online or through a Boston-area library system ([Minuteman Library Network](https://www.minlib.net/) or [Boston Public Library](https://www.bpl.org/)), as well as for purchase through your favorite bookseller.
+
+**NOTE**: This list was originally compiled by a non-beginner. If you are a beginner, we will be grateful for your [feedback](#edit-page).
+
+- [Think Python!](https://greenteapress.com/wp/think-python-2e/) by Allen Downey. This book starts from the most basic programming concepts and works toward a computer-science mindset as well as Python fluency. While no book is perfect for all readers, this one has found broad appeal with many beginners. The website has free PDF and HTML versions.
+
+- [Python Flash Cards](https://nostarch.com/pythonflashcards) by Eric Matthes. More like a deconstructed book, this box of 101 cards covers the basics of Python in a stripped-down fashion. These are not like traditional flash cards, which present a single concept per card. Each card is more like a quick reference for some piece of the larger Python puzzle. Potentially useful for beginners looking to review their understanding. Libraries: [MLN](https://find.minlib.net/iii/encore/record/C__Rb3814347).
+
+- [Automate the Boring Stuff With Python](https://automatetheboringstuff.com/) by Al Sweigart. A practical guide to the sorts of things that you can do with Python, how to do them, and why to do them that way. Recommended for those who have some understanding of Python and want to apply and improve their knowledge. Experienced programmers without previous Python exposure will probably be able to make good use of this book as well. The adventurous beginner might be able to make reasonable progress using the introductory chapter to learn Python fundamentals, but this is probably not an ideal place to start your Python journey. The book's content is available for free on the website. Libraries: [MLN](https://find.minlib.net/iii/encore/record/C__Rb3867472), [MLN <small>1<sup>st</sup> ed</small>](https://find.minlib.net/iii/encore/record/C__Rb3226226), [BPL <small>1<sup>st</sup> ed</small>](https://bpl.bibliocommons.com/item/show/6969316075).
```

Comment:
[Lots of good stuff in the new edition](https://inventwithpython.com/blog/2019/10/07/whats-new-in-the-2nd-edition-of-automate-the-boring-stuff-with-python/), but it seems like the 1st edition is still relevant.